### PR TITLE
Use generics for lookupFn

### PR DESCRIPTION
**Pull request by contexD**

### CODE REVIEW

File: index.test.ts [hunk 1]
```diff
@@ -23,13 +23,20 @@ describe('Test no cache', () => {
...
-        items.find((p) => p.slug === ref.slug),
+      lookupFn: (items: ArrayLike<MyValue | Error>, ref: MyKey) => {
+        for (let i = 0; i < items.length; i++) {
+          const myValue = isMyValue(items[i])
+          if (myValue && myValue.slug === ref.slug) {
+            return myValue
+          }
+        }
+        return undefined
+      },
```
Comment (mvantellingen) [inline index.test.ts]: Can't we keep using items.find() or is this more efficient?

Comment (contexD) [reply index.test.ts]: No, unfortunately not since the return type of the batchLoadFn is `ArrayLike` but not an actual array.

Comment (mvantellingen) [reply index.test.ts]: isn't this something: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#calling_find_on_non-array_objects ?

Comment (contexD) [reply index.test.ts]: Nice find! I've added it.

**Comment by mvantellingen** [comment]
Merged 👍

**Pull request closed without merging**